src/include/txn.i

```diff
@@ -791,12 +805,22 @@ __wt_txn_read_upd_list(WT_SESSION_IMPL *session, WT_UPDATE *upd, WT_UPDATE **upd
               F_ISSET(session, WT_SESSION_IGNORE_HS_TOMBSTONE) &&
               (upd->start_ts != WT_TS_NONE || upd->txnid != WT_TXN_NONE))
                 continue;
-            *updp = upd;
-            return (0);
+            break;
         }
         if (upd_visible == WT_VISIBLE_PREPARE)
             return (WT_PREPARE_CONFLICT);
     }
+    if (upd == NULL)
+        return (0);
+    /*
```

Comment:
I don't like the idea of reconstructing modify here. I think a lot of caller to this function doesn't care about the value and only want to know that the key is valid. Constructing modifies for these calls will be waste of effort.

Reply:
Can we still save the update and delay getting the value?

Reply:
I see your point. Specifically, some of the callers of `__wt_cursor_valid` are like this.

I'm trying to think of a good solution. I think having a `next` pointer on the `WT_UPDATE_VALUE` and reconstructed later would be bad because it's not consistent (i.e. history store modifies automatically get reconstructed already so it's a bit arbitrary if it doesn't get reconstructed just because it's an in-memory update). I'm thinking of having some parameter or flag to tell the read functions not to bother assigning the buffer and just set the update information like type, timestamp, etc to the `WT_UPDATE_VALUE`.

Reply:
fe831aa